### PR TITLE
MVL-589: Added PythonPip and PyTorch tutorials

### PR DESCRIPTION
**Pull request by sahehb**

### CODE REVIEW

File: mevislab.github.io/content/tutorials/thirdparty/pytorch/pytorchexample1.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+---
+# Example 1: Install PyTorch by using the PythonPip module
+
+## Introduction
+The module `PythonPip` allows you to install additional Python packages to be used in MeVisLab. You should not use the general Python *pip* command, because MeVisLab will not know these packages and they cannot be used in MeVisLab directly.
+
+The module either allows to install packages into the global MeVisLab installation directory, or into your defined user package. We will use the user package directory, because then the installed packages remain available in your packages even if you uninstall or update MeVisLab. In addition to that, no administrative rights are necessary if you install into a user directory.
+
+{{<alert class="warning" caption="Warning">}}
+Installing additional Python packages into MeVisLab by using the `PythonPip` module requires administrative rights if you do not install into a user package. In addition to that, the installed packages are removed when uninstalling MeVisLab.
```
Comment (mrbean-bremen) [inline mevislab.github.io/content/tutorials/thirdparty/pytorch/pytorchexample1.md:23]: .. if you  did install MeVisLab for all users

File: mevislab.github.io/content/tutorials/thirdparty/pytorch/pytorchexample1.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+Example for CUDA support:
+{{< highlight filename="Command: pip install" >}}
+```txt
+torch torchvision --index-url https://download.pytorch.org/whl/cu117
+
+```
+{{</highlight>}}
+
+{{<alert class="info" caption="Attention">}}
+If you are behind a proxy, you may have to set the **HTTP_PROXY** and **HTTPS_PROXY** environment variables to the hostname and port of your proxy. These are used by pip when accessing the internet.
```
Comment (mrbean-bremen) [inline mevislab.github.io/content/tutorials/thirdparty/pytorch/pytorchexample1.md:60]: proxy server

File: mevislab.github.io/content/tutorials/basicmechanisms/macromodules/pythonpip.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+## Introduction
+MeVisLab already comes with a lot of integrated third party software tools ready to use. Nevertheless it might be necessary to install additional Python packages for your specific needs. This example shows the usage of the `PythonPip` module in MeVisLab for installing additional Python packages.
+
+The `PythonPip` module allows to work with the Python package manager pip. It can be used to install Python packages into the site-packages of the MeVisLab Python installation.
+
+It provides the full Python package eco-system to MeVisLab, but it has several drawbacks!
+* On macOS:
+  * you can only install Python-only packages (packages that contain Python C-Extensions are not supported)
+* On Windows and Linux:
+  * packages can contain C-Extensions (since we use the same MSVC compiler resp. same GCC settings as Python 3 itself)
```
[the remark this answers is not in the document]
Comment (sahehb) [reply mevislab.github.io/content/tutorials/basicmechanisms/macromodules/pythonpip.md]: already resolved with Uwe

File: mevislab.github.io/content/tutorials/basicmechanisms/macromodules/pythonpip.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+MeVisLab already comes with a lot of integrated third party software tools ready to use. Nevertheless it might be necessary to install additional Python packages for your specific needs. This example shows the usage of the `PythonPip` module in MeVisLab for installing additional Python packages.
+
+The `PythonPip` module allows to work with the Python package manager pip. It can be used to install Python packages into the site-packages of the MeVisLab Python installation.
+
+It provides the full Python package eco-system to MeVisLab, but it has some drawbacks:
+
+* packages can contain C-Extensions (since we use the same MSVC compiler resp. same GCC settings as Python 3 itself)
+* *but* you can only install packages that do not interfere with packages or DLLs that are already part of MeVisLab
+* this means that installing packages with C-Extensions might work in many circumstances, but is not guaranteed to work
+* **all installed packages with C-Extensions are release only**, so you can only import them in a release MeVisLab (under Windows)
```
Comment (mrbean-bremen) [inline mevislab.github.io/content/tutorials/basicmechanisms/macromodules/pythonpip.md:26]: I find it a bit confusing that there are 4 bullet point for two drawbacks - the first 3 lines belong together

**Review by mrbean-bremen — APPROVED** [review]

File: mevislab.github.io/content/tutorials/thirdparty/pytorch/pytorchexample2.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+## Introduction
+In the previous tutorial, you learned how to install PyTorch for usage in MeVisLab. In this example we are using a trained network from torch.hub for creating a brain parcellation map on MeVisLab demo images.
+
+Make sure that you successfully executed [Example 1: Install PyTorch by using the PythonPip module](/tutorials/thirdparty/pytorch/pytorchexample1/) before starting.
+
+Add the module `PythonPip` to your workspace and make sure that *torch* and *torchvision* are installed in your MeVisLab package.
+
+## Steps to do
+Add a `PythonImage` module to your workspace. We are using it to load an image from disk and execute the PyTorch network. Create your own MeVisLab package as described in [Example 2.1: Package creation](/tutorials/basicmechanisms/macromodules/package/)
```
Comment (aliahcodes) [inline mevislab.github.io/content/tutorials/thirdparty/pytorch/pytorchexample2.md:25]: the disk

File: mevislab.github.io/content/tutorials/thirdparty/pytorch/pytorchexample2.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+Save your network containing the `PythonImage` module as *\*.mlab* file. Create a Macro Module as described in [Example 2.2: Global Macro modules](/tutorials/basicmechanisms/macromodules/globalmacromodules/) and use the network you just saved. Make sure to add a Python file in Macro Module Wizard.
```
Comment (aliahcodes) [inline mevislab.github.io/content/tutorials/thirdparty/pytorch/pytorchexample2.md:27]: "Make sure to add a Python file while creating the macro module through the Macro Module Wizard"

File: mevislab.github.io/content/tutorials/thirdparty/pytorch/pytorchexample2.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+Open the *\*.script* file of your Macro Module via right click {{< mousebutton "right" >}} and select {{< menuitem "Related Files" "<MODULE_NAME>.script">}}
+
+### Write the script file
```
Comment (aliahcodes) [inline mevislab.github.io/content/tutorials/thirdparty/pytorch/pytorchexample2.md:31]: "Content of the .script file" instead of current title

**Review by aliahcodes — CHANGES_REQUESTED** [review]
Request for changes in structure and language

File: mevislab.github.io/content/tutorials/basicmechanisms/macromodules/pythonpip.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+# Example 4: Install additional Python packages via PythonPip module
+## Introduction
+MeVisLab already comes with a lot of integrated third party software tools ready to use. Nevertheless it might be necessary to install additional Python packages for your specific needs. This example shows the usage of the `PythonPip` module in MeVisLab for installing additional Python packages.
+
+The `PythonPip` module allows to work with the Python package manager pip. It can be used to install Python packages into the site-packages of the MeVisLab Python installation.
+
+It provides the full Python package eco-system to MeVisLab, but it has some drawbacks:
+
+* packages can contain C-Extensions (since we use the same MSVC compiler resp. same GCC settings as Python 3 itself), *but* you can only install packages that do not interfere with packages or DLLs that are already part of MeVisLab. This means that installing packages with C-Extensions might work in many circumstances, but is not guaranteed to work
+* **all installed packages with C-Extensions are release only**, so you can only import them in a release MeVisLab (under Windows)
```
[the remark this answers is not in the document]
Comment (sahehb) [reply mevislab.github.io/content/tutorials/basicmechanisms/macromodules/pythonpip.md:24]: yes, Linux

File: mevislab.github.io/content/tutorials/thirdparty/pytorch.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+Some alternatives are:
+* [TensorFlow](https://www.tensorflow.org "tensorflow")
+* [Keras](https://keras.io "Keras")
+* [scikit-learn](https://scikit-learn.org "scikit-learn")
+
+## Available Tutorials
+### Install PyTorch by using the PythonPip module
+The first example shows how to install *torch* and *torchvision* by using the MeVisLab module `PythonPip`. This module can be used to install Python packages not integrated into MeVisLab.
+
+###
```
[the remark this answers is not in the document]
Comment (sahehb) [reply mevislab.github.io/content/tutorials/thirdparty/pytorch.md:30]: another headline still to come :)

**Pull request merged**